### PR TITLE
Upgrade to lodash 4

### PR DESCRIPTION
**Pull request by forivall**
Briefly discussed on slack.

One of the test fixtures was also caught by my grep/sed'ing, just to note. (I did double- & triple-check everything after the sed & grep, ran tests, etc.) I'll also check the coverage results to make sure all the changes were covered.

### CODE REVIEW

File: packages/babel-core/src/helpers/merge.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-import merge from "lodash/object/merge";
+import mergeWith from "lodash/mergeWith";
 
 export default function (dest?: Object, src?: Object): ?Object {
   if (!dest || !src) return;
 
-  return merge(dest, src, function (a, b) {
+  return mergeWith(dest, src, function (a, b) {
     if (b && Array.isArray(a)) {
       let newArray = b.slice(0);
```
Comment (forivall) [inline packages/babel-core/src/helpers/merge.js]: could the rest of this also be replaced with `_.union`?

File: packages/babel-core/test/util.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ suite("util", function () {
     assert.deepEqual(util.regexify(false), /.^/);
     assert.deepEqual(util.regexify(null), /.^/);
     assert.deepEqual(util.regexify(""), /.^/);
-    assert.deepEqual(util.regexify(["foo", "bar"]), /\x66oo|\x62ar/i);
+    assert.deepEqual(util.regexify(["foo", "bar"]), /foo|bar/i);
```
Comment (forivall) [inline packages/babel-core/test/util.js]: Please make sure that this change is correct.

**Comment by forivall** [comment]
I can't figure out what the CI issues are. I thought it might be an issue with npm v2, but I just ran the tests locally with node 3 and npm 2, and it works fine. Can they be re-run?

**Comment by forivall** [comment]
All of the tests pass on travis when running cleanly on my branch: https://travis-ci.org/forivall/babel

kcheck / eslint depends on lodash 3, and it seems to be cached in some deduplicated state. When it is restored from the cache, it doesn't specifically contain lodash, since it expecs it in the parent. But when bootstrap `npm install` is run, it changes the lodash version to 4, which breaks imports.

**Comment by himeshladva-ni** [comment]
@forivall Can this be progressed any further?

**Comment by billyjanitsch** [comment]
Hi @hzoo and @loganfsmyth. I'd love to see this Lodash bump get merged -- without it, projects that depend on Babel and lodash@4 can't dedupe a second version, so lodash@3 gets installed dozens of times for a typical Babel config.

Would it be helpful for me to rebase this PR against master? Are there any other blockers that I can clear up?

**Comment by hzoo** [comment]
Not a big deal for node 4, etc, then. @loganfsmyth

**Comment by hzoo** [comment]
Thanks so much!

**Pull request merged**